File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var numSubarrayProductLessThanK = function(nums, k) {
    let product=1;
    const n = nums.length;
    let i=0;
    let j=0;
    if (k==0 || k==1){
        return 0;
    }
    let count=0;

    while(j<n){
        product= product*nums[j];
        while (product>=k){
            product = product / nums[i];
            i++;
        }
        count = count + (j-i+1)
        j++;
    }
    return count;
};

File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        int i=0;
        int j=0;
        int count=0;
        int p=1;
        int n = nums.size();
        if(k==0)  return 0;
        while(j<n){
            p= p*nums[j];
            while(p>=k){
                p= p/nums[i];
                i++;
            }
            count = count + (j-i+1);
            j++;
        }
        return count;
    }
};